src/core/caniuse.js

```diff
@@ -0,0 +1,212 @@
+/*
+Module: "core/caniuse"
+Adds a caniuse support table for a "feature" #1238
+
+`conf.caniuse = { feature: key }` =>
+  .. the table is added before Copyright
+Optional settings:
+  `conf.caniuse.browsers` list of caniuse supported browser names
+    to be shown in the table or "ALL"
+    default: ["chrome", "firefox", "safari", "edge"]
+  `conf.caniuse.versions` number of browser versions to show
+  `conf.caniuse.maxAge` (in ms) local response cache duration
+  `conf.caniuse.apiURL` URL from where to fetch stats.
+    use {FEATURE} as placeholder in URL to replace it by a feature name
+*/
+
+import { semverCompare } from "core/utils";
+import IDBCache from "core/idb-cache";
+import { pub } from "core/pubsubhub";
+import "deps/hyperhtml";
+import caniuseCss from "deps/text!core/css/caniuse.css";
+
+export const name = "core/caniuse";
+
+export async function run(conf) {
+  if (!conf.caniuse) {
+    return;
+  }
+  // Opportunistically insert the style into the head to reduce FOUC.
+  const codeStyle = document.createElement("style");
+  codeStyle.textContent = caniuseCss;
+  document.head.appendChild(codeStyle);
+
+  normalizeConf(conf);
+  const { caniuse } = conf;
+  const cache = await new IDBCache("respec-caniuse", ["caniuse"]);
+  if (caniuse.feature) {
+    canIUse(
+      caniuse.feature,
+      document.querySelector(".head dl").lastChild,
+      cache,
+      caniuse
+    );
+  }
+}
+
+/**
+ * MUTATES `conf.caniuse` object to hold normalized configurarion
+ * @param {Object} conf   configuration settings
+ */
+function normalizeConf(conf) {
```

Comment:
We can optimize this a little bit as follows. Also, we should do the browser filtering/checks here, and just emit warnings:

```JS
function normalizeConf(conf) {
  const DEFAULTS = {
    maxAge: 24 * 60 * 60 * 1000, // 24 hours (in ms)
    browsers: ["chrome", "firefox", "safari", "edge"],
    versions: 4,
  };
  if (typeof conf.caniuse === "string") {
    conf.caniuse = { feature: conf.caniuse, ...DEFAULTS };
    return;
  } else {
    conf.caniuse = { ...DEFAULTS, ...conf.caniuse };
  }
  if (Array.isArray(conf.caniuse.browsers)) {
    conf.caniuse.browsers = conf.caniuse.browsers
      .map(b => b.toLowerCase())
      .filter(b => { /*
          warn here if invalid value
      */});
  } else if (conf.caniuse.browsers !== "ALL") {
    conf.caniuse.browsers = [...DEFAULTS.browsers];
  }
}
```